core/mondoo-ms365-security.mql.yaml

```diff
@@ -0,0 +1,611 @@
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          microsoft.policies.identitySecurityDefaultsEnforcementPolicy["isEnabled"] == false
+          ```
+      remediation: |
+        ### Microsoft 365 Console
+         
+         To update via the Microsoft 365 portal:
+         1. Log in as Global Administrator, security administrator or Conditional Access administrator to the Microsoft 365 portal at https://admin.microsoft.com
+         2. Select "Azure Active Directory" once in the "Admin Centers"
+         3. Overview --> properties --> Manage Security default -> Verify the Enabled security default toggle to no 
+         4. Save     
+    query: |
+      microsoft.policies.identitySecurityDefaultsEnforcementPolicy["isEnabled"] == false
+  - uid: mondoo-m365-security-ensure-that-spf-records-are-published-for-all-exchange-domains
+    title: Ensure Sender Policy Record (SPF) exists for each domain in Exchange 
+    severity: 60
+    docs:
+      desc: |
+        This control checks that SPF records is created for each domain in Exchange. 
+      audit: |
+        __cnquery run__
+
+        To audit Microsoft 365 with `cnquery run`:
+
+        Run the following query:
+
+          ```mql
+          cnquery run ms365 -c "microsoft.domains { serviceConfigurationRecords.one(recordType.downcase == "txt" && properties['text'] == "v=spf1 include:spf.protection.outlook.com -all") }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        __cnquery shell__
+
+        To audit Microsoft 365 with `cnquery shell`:
+
+        1. Launch `cnquery shell`:
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          microsoft.domains { serviceConfigurationRecords.one(recordType.downcase == "txt" && properties['text'] == "v=spf1 include:spf.protection.outlook.com -all") }
+          ```
+      remediation: |
+        If Exchange Online is the main platform for sending and receiving the email in your domain, add the following txt record for each accepted domains:
+          v=spf1 include:spf.protection.outlook.com -all 
+    query: |
+      microsoft.domains {
+        serviceConfigurationRecords.one(recordType.downcase == "txt" && properties['text'] == "v=spf1 include:spf.protection.outlook.com -all")
+      }
+  - uid: mondoo-m365-security-ensure-that-mobile-devices-require-a-minimum-password-length-to-prevent-brute-force-attacks
+    title: Ensure that minimum password length has been set in order to prevent any brute force attack 
+    severity: 80
+    docs:
+      desc: |
+        This control checks that there is minmum password length - at least six characters - for mobile devices 
```

Comment:
should we go with 8. https://pages.nist.gov/800-63-3/sp800-63b.html

Reply:
Thanks Patrick! Very good point. Changed it to 8 and added following line from the document you referred!

According to NIST (SP 800-63-2), permitted the use of randomly generated PINs with 6 or more digits while requiring user-chosen memorized secrets to be a minimum of 8 characters long.